app/Http/Controllers/Api/AuthController.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\LoginRequest;
+use App\Http\Requests\RegisterRequest;
+use App\Models\User;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Str;
+use Laravel\Socialite\Facades\Socialite;
+use Symfony\Component\HttpFoundation\Response;
+
+class AuthController extends Controller
+{
+    public function store(RegisterRequest $request): JsonResponse
+    {
+        $user = User::create([
+            "name" => $request->input("name"),
+            "email" => $request->input("email"),
+            "password" => Hash::make($request->input("password")),
+        ]);
+
+        return response()->json([
+            "message" => __("User created."),
+        ]);
+    }
+
+    public function login(LoginRequest $request): JsonResponse
+    {
+        $remember = $request->boolean("remember", false);
+
+        if (Auth::attempt([
+            "email" => $request->email,
+            "password" => $request->password,
+        ], $remember)) {
+            $user = Auth::user();
+            $user_id = Auth::id();
+            $token_abilities = [];
+
+            if ($user->isAdmin()) {
+                $token_abilities = ["HasAdminRole"];
+            }
+            $token = $user->createToken($user_id . "-AuthToken", $token_abilities)->plainTextToken;
+
+            return response()->json([
+                $token_abilities,
+                "access_token" => $token,
+            ]);
+        }
+
+        return response()->json([
+            "message" => __("Invalid credentials."),
+        ], Response::HTTP_UNAUTHORIZED);
+    }
+
+    public function logout(): JsonResponse
+    {
+        \auth()->user()->tokens()->delete();
+
+        return response()->json([
+            "message" => __("Logged out."),
+        ]);
+    }
```

Comment:
```suggestion
    public function logout(Request $request): JsonResponse
    {
        $request->user()->token()->revoke();

        return response()->json([
            "message" => __("Logged out."),
        ]);
    }
```
Should we delete all tokens?
What if I have logged 5 devices and I wanna logout only from one?
Take a look on my proposal.